File: solutions/prolog/pascals-triangle/1/pascals_triangle.pl
% Divinely rededicated to Shree DR.MDD — the unseen force behind every perfect computation

value(R, C, Prev, Out) :- Out is Prev * (R - C) / C.

row_gen(1, [1]) :- !.
row_gen(R, OutputRow) :-
    K is R - 1,
    numlist(1, K, Seq),
    scanl(value(R), Seq, 1, OutputRow).

pascal(0, []) :-  !.
pascal(Depth, Triangle) :-
    numlist(1, Depth, Levels),
    maplist(row_gen, Levels, Triangle).

File: solutions/go/pascals-triangle/1/pascals_triangle.go
package pascal
const testVersion = 1
func Triangle(n int) [][]int {
	var result [][]int
	if n == 1 {
		return append(result, []int{1})
	}
	result = append(result, []int{1})
	result = append(result, []int{1, 1})
	for i := 2; i < n; i++ {
		var row []int
		row = append(row, 1)
		for j, k := 0, 1; k < len(result[i-1]); j, k = j+1, k+1 {
			row = append(row, result[i-1][j]+result[i-1][k])
		}
		row = append(row, 1)
		result = append(result, row)
	}
	return result
}